sys/key_rev_large_value.js

```diff
@@ -0,0 +1,449 @@
+        ]
+    };
+};
+
+ChunkedBucket.prototype._makeChunksSchema = function(opts) {
+    var schemaVersionMajor = 1;
+
+    return {
+        version: schemaVersionMajor * 1000 + (opts.version || 0),
+        options: {
+            compression: opts.compression || [
+                {
+                    algorithm: 'deflate',
+                    block_size: 1024
+                }
+            ]
+        },
+        attributes: {
+            key: opts.keyType || 'string',
+            chunk_id: 'int',
+            tid: 'timeuuid',
+            value: opts.valueType || 'blob'
+        },
+        index: [
+            { attribute: 'key', type: 'hash' },
+            { attribute: 'chunk_id', type: 'hash' },
+            { attribute: 'tid', type: 'range', order: 'desc' }
+        ]
+    };
+};
+
+ChunkedBucket.prototype._validateSchemaOptions = function(req) {
+    var options = req.body || {};
+
+    if (options.retention_policy) {
+        throw new Error('key_rev_large_value does not support revision_policy option');
+    }
+
+    if (options.compression) {
+        throw new Error('key_rev_large_value does not support compression option');
+    }
+};
+
+ChunkedBucket.prototype.createBucket = function(hyper, req) {
+    this._validateSchemaOptions(req);
+    var metaSchema = this._makeMetaSchema(req.body || {});
+    var dataSchema = this._makeChunksSchema(req.body || {});
+    metaSchema.table = this._metaTableName(req);
+    dataSchema.table = this._chunksTableName(req);
+    return P.join(
+        hyper.put({
+            uri: new URI([req.params.domain, 'sys', 'table', metaSchema.table]),
+            body: metaSchema
+        }),
+        hyper.put({
+            uri: new URI([req.params.domain, 'sys', 'table', dataSchema.table]),
+            body: dataSchema
+        }))
+    .then(function() { return { status: 201 }; });
+};
+
+// Format a revision response. Shared between different ways to retrieve a
+// revision (latest & with explicit revision).
+function returnRevision(req, metadata) {
+    return function(dbResult) {
+        if (dbResult.body && dbResult.body.items && dbResult.body.items.length) {
+            var row = dbResult.body.items[0];
+            return {
+                status: 200,
+                headers: metadata.headers,
+                body: row.value
+            };
+        } else {
+            throw new HTTPError({
+                status: 404,
+                body: {
+                    type: 'not_found',
+                    uri: req.uri,
+                    method: req.method
+                }
+            });
+        }
+    };
+}
+
+function coerceTid(tidString) {
+    if (uuid.test(tidString)) {
+        return tidString;
+    }
+
+    // Out of luck
+    throw new HTTPError({
+        status: 400,
+        body: {
+            type: 'key_rev_value/invalid_tid',
+            title: 'Invalid tid parameter',
+            tid: tidString
+        }
+    });
+}
+
+function parseRevision(rev) {
```

Comment:
This smells like a good candidate to be moved to `mwUtil`. `coerceTid()` too actually.